src/dwarf.cpp

```diff
@@ -1109,6 +1109,57 @@ attribute_value dwarf::implementation::process_form(const attribute& attr,
         result.reference(static_cast<std::uint32_t>(cu_offset + offset));
     };
 
+    const auto handle_passover = [&](){
+        // We have a problem if we are passing over an attribute that is needed to determine
+        // ODRVs.
+        assert(nonfatal_attribute(attr._name));
+        result.passover();
+        auto size = form_length(attr._form, _s);
+        _s.seekg(size, std::ios::cur);
+    };
+
+    enum class block_type {
+        one,
+        two,
+        four,
+        uleb,
+    };
+
+    const auto handle_block = [&](std::size_t size) {
+        if (size > 8) {
+            throw std::runtime_error("Unexpected block size read of essential data.");
+        }
+
+        std::uint64_t value(0);
+        while (size--) {
+            value <<= 8;
+            value |= read8();
+        }
+
+        result.uint(value);
+    };
+
+    const auto maybe_handle_block = [&](block_type type){
```

Comment:
The first level of block handling - if the attribute is nonessential, we pass over it like we were doing before. If it is essential, depending on the `blockN` form, we read some number of bytes to discover how much data this block holds. We then forward that size on to `handle_block`, above.